### PR TITLE
[19268] Replace uint64_t by 8 in alignas specifier

### PR DESCRIPTION
**Pull request by rsanchez15**
## Description

Some android build targets cannot translate from `alignas(uint64_t)` to `alignas(8)`.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/rtps/DataSharing/DataSharingNotification.hpp [hunk 1]
```diff
@@ -93,7 +93,7 @@ class DataSharingNotification
 
 #pragma warning(push)
 #pragma warning(disable:4324)
-    struct alignas (uint64_t) Notification
+    struct alignas (8) Notification
```
Comment (jsantiago-eProsima) [inline src/cpp/rtps/DataSharing/DataSharingNotification.hpp:96]: Maybe using something like `alignas ( sizeof (uint64_t) )` would be cleaner and clearer.

File: src/cpp/rtps/DataSharing/DataSharingPayloadPool.hpp [hunk 2]
```diff
@@ -142,7 +142,7 @@ class DataSharingPayloadPool : public IPayloadPool
 
 #pragma warning(push)
 #pragma warning(disable:4324)
-    class alignas (uint64_t) PayloadNode
+    class alignas (8) PayloadNode
```
Comment (jsantiago-eProsima) [inline src/cpp/rtps/DataSharing/DataSharingPayloadPool.hpp:145]: Same comment.

File: src/cpp/rtps/DataSharing/DataSharingPayloadPool.hpp [hunk 3]
```diff
@@ -338,7 +338,7 @@ class DataSharingPayloadPool : public IPayloadPool
 
     };
 
-    struct alignas (uint64_t) PoolDescriptor
+    struct alignas (8) PoolDescriptor
```
Comment (jsantiago-eProsima) [inline src/cpp/rtps/DataSharing/DataSharingPayloadPool.hpp:341]: Same comment.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Review by jsantiago-eProsima — APPROVED** [review]

**Comment by EduPonz** [comment]
@Mergifyio backport 2.6.x 2.10.x 2.11.x

**Pull request merged**